### DISCUSSION TITLE
Technical OGame source

### DISCUSSION THREAD

**Discussion by ghost** [lanedirt/OGameX#312] [General]
Hi @lanedirt,

I wanted to share the primary source from which I derive most of my information regarding OGame formulas. As a former French OGame player, we've worked hard over time to ensure that the game operators maintain this thread as accurate as possible.

Please note that the content is in French, but feel free to use translators if you don't speak the language if this might be of any help.

You can find the thread here: [OGame Formulas & Options (French)](https://board.fr.ogame.gameforge.com/index.php?thread/710820-options-formules/&postID=11765361#post11765361).

**NB:** I am using this code base, as mentioned earlier, for an Android application in a private repository. When syncing my private repository with the public OGameX repository, I always need to filter out content I've changed that is not related to OGameX. Most of the content is developed for the API and is excluded as it are new controllers in the api directory, so it’s not a problem. However, some original files have been edited to integrate Sanctum for the API part.

Would you mind if I share the API singleton I used? This would help me avoid having to filter out changes every time I commit to the public repository.

**Comment by lanedirt** [comment]
Hi @Jbaukens,

Thanks for sharing the URL, good to have an overview of all business rules in one place! Will come in handy for sure!

For the API additions, I'm open to the idea of including generic base modules into the OGameX repo that your fork depends on. As long as they are generic I don't see the harm in that.

I did not have any plans for OGameX to expose API's (yet) as there is still quite some work to do on the base game. But I'm not against changing/adding logic to the core so it can (officially) support an API structure in the future. So feel free to open a PR with the proposed changes and I'll be happy to take a look at it.
